### DISCUSSION TITLE
How to ensure a formula is added to my path after installation?

### DISCUSSION THREAD

**Discussion by Southpaw1496** [Homebrew/discussions#3915] [Writing Formulae/Casks]
I'm making a formula for the [Quilt Native Installer](https://github.com/quiltmc/quilt-native-installer). It's a GUI program that's distributed as a Rust executable, so I want to be able to run it by typing `quilt` in the terminal, instead of keeping it somewhere like `/Applications`, where it doesn't really belong. Here's my formula file, still a work in progress:

```rb
# Documentation: https://docs.brew.sh/Formula-Cookbook
#                https://rubydoc.brew.sh/Formula
# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
class Quilt < Formula
  desc ""
  homepage ""
  url "https://github.com/QuiltMC/quilt-native-installer/archive/refs/tags/v0.1.1.tar.gz"
  sha256 "42d80580dc11e3a883b06633450f8121de704cb0ba24ac56d067f898473eddaa"
  license "Apache-2"

  depends_on "rust" => :build

  def install
    # ENV.deparallelize  # if your formula fails when building in parallel
    system "cargo", "install", *std_cargo_args
  end

  test do
    # `test do` will create, run in and delete a temporary directory.
    #
    # This test will fail and we won't accept that! For Homebrew/homebrew-core
    # this will need to be a test that verifies the functionality of the
    # software. Run the test with `brew test quilt`. Options passed
    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
    #
    # The installed folder is not in the path, so use the entire path to any
    # executables being tested: `system "#{bin}/program", "do", "something"`.
    system "false"
  end
end
```

I have built and installed it successfully using `brew install --build-from-source southpaw1496/homebrew/quilt`, and it shows in `brew leaves`, but it isn't in my path. What do I need to change so that it is?

**Comment by SMillerDev** [comment]
Is there something in `$(brew --prefix)/opt/quilt/bin` after?

**Reply by Southpaw1496** [reply]
Ah yes, there's `quilt-installer`, which is in my path. Can I change the name of the executable without modifying the source code?

**Reply by SMillerDev** [reply]
I think cargo defines that normally but I guess you could rename the binary in the formula.

**Reply by Southpaw1496** [reply]
I tried to use `bin.install "quilt-installer" => "quilt"` but after trying that, it turns out the file that gets built is `quilt-native-installer-<tag>` (e.g `quilt-native-installer-0.1.1`), which means that it will change every time the formula updates. Manually updating the `bin.install line` every time I need to update it seems like bad practice, so is there any way I can substitute the name of Cargo's outputted binary into the command?

**Reply by SMillerDev** [reply]
Well, there is`#{version}`. But I think this should be configurable in cargo.
